lib/streamlit/type_util.py

```diff
@@ -197,13 +197,15 @@ def get_fqn_type(obj: object) -> str:
 _PANDAS_SERIES_TYPE_STR: Final = "pandas.core.series.Series"
 _PANDAS_STYLER_TYPE_STR: Final = "pandas.io.formats.style.Styler"
 _NUMPY_ARRAY_TYPE_STR: Final = "numpy.ndarray"
+_SNOWPARK_DF_TYPE_STR: Final = "snowflake.snowpark.dataframe.DataFrame"
 
 _DATAFRAME_LIKE_TYPES: Final[tuple[str, ...]] = (
     _PANDAS_DF_TYPE_STR,
     _PANDAS_INDEX_TYPE_STR,
     _PANDAS_SERIES_TYPE_STR,
     _PANDAS_STYLER_TYPE_STR,
     _NUMPY_ARRAY_TYPE_STR,
+    _SNOWPARK_DF_TYPE_STR,
```

Comment:
Why is this needed in this PR? `st.memo` does not use `is_dataframe_like`, or?

Reply:
Rebase mistake, it's reverted now 👍